### ISSUE TITLE
[Feature Request] update Matroska file with edited subtitles

### ISSUE THREAD

**Issue by mtarini** [SubtitleEdit/subtitleedit#3279]
First of all, GREAT application, big long-term fan here. 
The range of well designed and well implemented features is amazing!

The suggested feature:
support for **updating** Matroska files with the edited subtitles.  

I understand that Matroska file format can potentially include a great deal of diverse stuff, and that SubtitleEdit should not try to become an editor for that, or a packager for new Matroska files.  

Still, SubtitleEdit could cover just the **most typical case**, namely, the pipeline: 

1.  **import subtitles from MKV**

2. **edit them**, in whichever way

3. **export them back** <== _this is the missing part_
That is: 
_substitute the original imported subtitles with the edited one, 
while leaving anything else in the MKV file unchanged._

This could be accessible to the user as a new "Update current Matroska file..." command, under File menu, which is only available when the current subtitle file was imported from a .mkv.

**Comment by OmrSi** [comment]
I don't speak for @niksedk but I don't think he wants to add any video editing abilities, someone offered him code for cutting videos and other stuff and someone requested chapters editing abilities and he seems to only want SE to edit subs, which is better in my opinion.

You have mkvtoolnix for those things.
Anyway, he has the last word, this is just what I felt...

**Comment by mtarini** [comment]
Just as a clarification...

I totally agree with the above, i.e. that SE should not cover video editing / MKV packaging.

That's why I am proposing only a **minimal** functionality, which covers only the simplest, most typical case.

Specifically, the case when you edit a single subtitle-track from a MKV file, and you want to store it back exactly where it was, with no other change in the MKV file or anything else.

That covers 99.5% of the cases (for me).  To use a second tool for that is always possible, but a bit cumbersome (and it feels overkill).

BTW, the same could be said for the current ability of SE to _extract_ subtitles from MKV files. Technically, there is no need for that: one could perfectly well use mkvtoolnix to extract the subtitle track, _then_ edit it with SE. But, it is so much more convenient to do this with SE alone! 
I am proposing the mirror functionality for saving, for the same reason.

(this functionality would be available only when the current subtitle track was extracted from a MKV file).

**Comment by mkver** [comment]
Most edits of subtitles will change the number of bytes needed for the subtitle block. This means that it is not enough to edit the subtitle block alone, you have to move everything that comes after it around. And you have to update the size fields of a lot of other elements (namely the segment containing the clusters and the clusters that contain the blocks). And then you would also have to update the cues (the index) because you moved the elements that they are pointing to. And one should also update the tags if they exist. And if the file uses CRC-checksums (as ffmpeg does by default), these need to be updated, too.

And if you also change the timing , then you should also change the interleavement (in a Matroska file, audio, video and subtitles for a certain timestamp are stored near to each other so that one can play the file without unnecessary seeks; if you edit the timing, you should actually also move the subtitles).

In short: It is not possible to simply edit something in a Matroska file, because one normally hasn't reserved space for it. You'd have to add a more or less complete Matroska muxer to fullfil your request. You'd better stick to MKVToolNix for this.
